metrics/prometheus-metrics/src/main/scala/sttp/tapir/server/metrics/prometheus/PrometheusMetrics.scala

```diff
@@ -48,16 +49,18 @@ case class PrometheusMetrics[F[_]](
 
 object PrometheusMetrics {
 
-  implicit val schemaForCollectorRegistry: Schema[CollectorRegistry] = Schema.string[CollectorRegistry]
+  implicit val schemaForPrometheusRegistry: Schema[PrometheusRegistry] = Schema.string[PrometheusRegistry]
 
-  implicit val collectorRegistryCodec: Codec[String, CollectorRegistry, CodecFormat.TextPlain] =
-    Codec.anyString(TextPlain())(_ => DecodeResult.Value(new CollectorRegistry()))(r => {
-      val output = new StringWriter()
-      TextFormat.write004(output, r.metricFamilySamples)
+  implicit val prometheusRegistryCodec: Codec[String, PrometheusRegistry, CodecFormat.TextPlain] =
+    Codec.anyString(TextPlain())(_ => DecodeResult.Value(new PrometheusRegistry()))(r => {
+      val output = new ByteArrayOutputStream()
+      ExpositionFormats.init().getPrometheusTextFormatWriter.write(output, r.scrape())
```

Comment:
is it ok to call `ExpositionFormats.init()` on every metrics request, or maybe this should be done once?

Reply:
good idea. I will move that to a `val`